ui/org.eclipse.pde.junit.runtime/src/org/eclipse/pde/internal/junit/runtime/RemotePluginTestRunner.java

```diff
@@ -99,19 +104,38 @@ private static ClassLoader createJUnit5PluginClassLoader(String testPluginName)
 
 	private static List<Bundle> findTestEngineBundles() {
 		BundleContext bundleContext = FrameworkUtil.getBundle(RemotePluginTestRunner.class).getBundleContext();
-		List<Bundle> engineBundles = new ArrayList<>();
-		for (Bundle bundle : bundleContext.getBundles()) {
-			try {
-				BundleWiring bundleWiring = bundle.adapt(BundleWiring.class);
-				Collection<String> listResources = bundleWiring.listResources("META-INF/services", "org.junit.platform.engine.TestEngine", BundleWiring.LISTRESOURCES_LOCAL); //$NON-NLS-1$//$NON-NLS-2$
-				if (!listResources.isEmpty()) {
-					engineBundles.add(bundle);
-				}
-			} catch (Exception e) {
-				// check the next bundle
+		return Arrays.stream(bundleContext.getBundles()).filter(RemotePluginTestRunner::providesCompatibleTestEngine).collect(toList());
```

Comment:
Using `Collectors.toCollection(ArrayList::new)` would avoid the need to change the caller
```suggestion
		return Arrays.stream(bundleContext.getBundles()).filter(RemotePluginTestRunner::providesCompatibleTestEngine).collect(toCollection(ArrayList::new));
```